tests/31sync/05presence.pl

```diff
@@ -51,13 +49,25 @@
       })->then( sub {
          my ( $body ) = @_;
 
+         # Either I'm absent entirely, or I'm present in state "offline"; but
+         # I shouldn't be "online"
+
          my $events = $body->{presence}{events};
-         my $presence = first { $_->{type} eq "m.presence" } @$events
-            or die "Expected to see our own presence";
 
-         $presence->{sender} eq $user->user_id or die "Unexpected sender";
-         $presence->{content}{presence} eq "offline"
-            or die "Expected to be offline";
+         if( my $presence = first { $_->{type} eq "m.presence" } @$events ) {
```

Comment:
Surely the $presence->{sender} eq $user->user_id check should be in this filter? It may possibly be valid for other users' presence to be reported.

(Maybe we would auto-populate other people on your team/company/the whole server/whatever into your presence list, in the future. Or Myspace Tom.)